content/shmem_symmetric_available.tex

```diff
@@ -0,0 +1,33 @@
+\apisummary{
+Returns the memory available on the symmetric heap in bytes.}
+
+
+\begin{apidefinition}
+
+\begin{Csynopsis}
+void shmem_symmetric_available(size_t *available_size);
+\end{Csynopsis}
+
+\begin{apiarguments}
+\apiargument{OUT}{available\_size}{The symmetric heap memory available in bytes.}
+\end{apiarguments}
+
+
+\apidescription{
+The \FUNC{shmem\_symmetric\_available} call returns the memory available on the symmetric heap in bytes.
+The available memory is the unallocated memory on the symmetric heap, which could be allocated by using memory allocation routines such as \FUNC{shmem\_malloc} or \FUNC{shmem\_calloc}. 
```

Comment:
@bcernohous I disagree. There are many scenarios where we want to allocate all of available symmetric memory heap memory. One scenario where we encountered this - using the symmetric heap as a cache for I/O workloads. Here, we wanted to use  all of available 
heap as a cache. If you want to do something like this, with the current specification, you have do the bookkeeping in the application to figure out the available memory. 

I’m not disputing that more details are useful. If there is clear motivation, we can add it.